### PR TITLE
Add documentation for `dynamical.ops` and `dynamical.handlers`

### PR DESCRIPTION
**Pull request by azane**
Partly addresses #337 by adding documentation for `simulate`, `LogTrajectory` `StaticBatchObservation` and all user-facing interruptions.

### CODE REVIEW

File: docs/source/type_aliases.rst
Comment (azane) [inline docs/source/type_aliases.rst:1]: Not sure if we actually want this, but this was the cleanest way I could find (after looking only a little bit) to document these type aliases. Happy to remove if not desired.

**Comment by SamWitty** [comment]
@azane , could you address the lint failure? Thanks!

**Comment by SamWitty** [comment]
Confirming that operations and handlers render correctly when I build locally:

Screenshot 2023-12-08 at 10 57 11 AM

However, as my comment above about the type aliases warning implies, I don't think the type aliases are currently included in the docs build.

**Comment by SamWitty** [comment]
> One thing I notice in the above rendering is that the `State` type alias has been evaluated away. E.g. rather than `State[T]` it shows up as `Mapping[str, T]`. This is maybe fine, but it will make things confusing if we use the type alias name directly in the documentation itself. The example I showed above has `State[T]` listed as the return type.

@azane and I figured out a workaround, but will plan on submitting a follow-up
PR as this issue is pervasive throughout all of the documentation, not just dynamical systems.

**Pull request merged**